src/modules/edc-demo/components/contract-agreement-transfer-dialog/model/http-datasink-form-mapper.ts

```diff
@@ -0,0 +1,51 @@
+import {Injectable} from '@angular/core';
+import {HttpDatasinkProperties} from '../../../models/http-datasink-properties';
+import {ContractAgreementDatasinkFormValue} from './contract-agreement-datasink-form-model';
+import {HttpDatasinkHeaderFormValue} from './http-datasink-header-form-model';
+
+@Injectable({providedIn: 'root'})
+export class HttpDatasinkFormMapper {
+  buildHttpDatasinkProperties(
```

Comment:
this would be buildHttpRequestParams then

Reply:
If you're smart you could create a type that will help you deal with both forms at once, making it

```ts
/**
 * All common fields of Http Datasource Form Value and Http Datasink Form Value.
 */
type HttpRequestFormFields = 
  | 'httpMethod'
  | 'httpUrl' 
  | ... ;

/**
 * Common form value of Http Datasource and Http Datasink.
 * This is possible because in the Core EDC both are supported by HttpRequestParams.
 */
type HttpRequestParamsFormValue = 
  | Pluck<ContractAgreementTransferDialogForm, HttpRequestFormFields> 
  | Pluck<AssetDatasourceFormValue, HttpRequestFormFields>;

export function buildHttpRequestParams(formValue: HttpRequestParamsFormValue): HttpRequestParams {
```

(This is the power of typescript, to unify things that are separate, as long as it works)

I hope the above code works, just try it real quick, because if it does, it would allow us to not have to duplicate this very important validation / mapping code.

If it doesn't after a quick check, we'll keep it how you copied it.